### PR TITLE
e2e: Support any storage class name

### LINKED ISSUE #1635: e2e: Improve storage class names and supported storage checks
The e2e framework uses a list of storage class names and access modes:

```yaml
pvcspecs:
  - storageclassname: rook-cephfs
    accessmodes: ReadWriteMany
  - storageclassname: rook-ceph-block
    accessmodes: ReadWriteOnce
```

Based on this list, we generate test names:

```go
func generateSuffix(storageClassName string) string {
	suffix := storageClassName

	if strings.ToLower(storageClassName) == "rook-ceph-block" {
		suffix = "rbd"
	}

	if strings.ToLower(storageClassName) == "rook-cephfs" {
		suffix = "cephfs"
	}

	return suffix
}
```

This works only for drenv environment. For openshift environment the storage class names are different so the returned suffix with be the actual storage class name, which is way to long to use as a test name or for using resource names.

Even worse, there is code hidden deployers/discoveredapps.go, filtering out cephfs workloads for discovered apps, based on the suffix:

```go
func (d DiscoveredApps) IsWorkloadSupported(w workloads.Workload) bool {
	return w.GetName() != "Deploy-cephfs"
}
```

This will not match for when using real cluster storage classes, trying to run unsupported combination.

## Possible fix

Move the logic to the config:

```
storage:
  - name: file
    storageclassname: rook-cephfs
    accessmodes: ReadWriteMany
    supports-discovered-apps: false
  - name: block
    storageclassname: rook-ceph-block
    accessmodes: ReadWriteOnce
    supports-discovered-apps: true
```

- With this layout we can define any storage class and enable the storage class for discovered apps without changing the code. 
- We don't need the code mapping the actual storage class names to test names, instead we can use the name property.

Blocks #1625

### PR DESCRIPTION
**Pull request by nirs**
Fix the way we configure storage classes so we can create configuration that works with any clsuter.

- Replace coded using hardcoded storage class names with configuration
- Fix the way we filter unsupported workloads and deployers

Fixes #1635

### CODE REVIEW

File: e2e/config.yaml [hunk 1]
```diff
@@ -9,3 +9,5 @@ pvcspecs:
   - name: cephfs
     storageclassname: rook-cephfs
     accessmodes: ReadWriteMany
+    unsupportedDeployers:
```
Comment (raghavendra-talur) [inline e2e/config.yaml:12]: I prefer limiting the config file to user preferences. Whether a storageClass is not supported by a deployer is a function of the code not config.

**Comment by raghavendra-talur** [comment]
@nirs the move of the supported calls from deployer to workload hints at a problem in our framework. In the end it is the tuple (workload, deployment) which needs to be inspected for support status. It doesn't matter which entity inspects another, both seem wrong to me.

We should have helper functions which need not be methods on any type or even part of a interface which analyze a combination of workload and deployer and tell if it is a supported config.

File: e2e/config.yaml (hunk 1 above)
Comment (raghavendra-talur) [reply e2e/config.yaml:12]: So in combination with my other comment, the helper function can decide in code if a pvcspec name is incompatible with a workload,deployer config.

Comment (nirs) [reply e2e/config.yaml:12]: The code does not have any way to tell if something is supported or not. This is pure configuration.

When this is set in the configuration, I can create multiple configurations. For If I work on rbd related stuff, I can run the test only with rbd, without using cryptic -run arguments. This is much more powerful that doing this in code.

**Comment by nirs** [comment]
I rebased on main since it should not need the e2e-namesspace branch.

Keeping unsupported as configuration is important for upstream. Think about a vendor adapting e2e test for its storage type that we cannot even test (e.g. KubeSAN). We don't want to keep code knowing about all kinds of storage. This should be part of vendor configuration that may be kept upstream for easier maintenance. The way to make this work is to keep the knowledge in the configuration, and keep the code simple.

It is also very useful for developers that want run all the tests with specific storage. It is easier to keep multiple configurations than remember to run the tests with the right -run argument. And we need to manage multiple configuration anyway (e.g. drenv, your ocp test cluster).

This is same as storageClassName - we can use any value from the config. We can run the tests with 1 storageClassName or 4, without changing any code.

**Review by raghavendra-talur — APPROVED** [review]

**Pull request merged**